### PR TITLE
Add stubs for dateutil.relativedelta

### PR DESCRIPTION
**Pull request by rowillia**

### CODE REVIEW

File: third_party/2.7/dateutil/relativedelta.pyi [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Union
+from datetime import date, datetime, timedelta
+
+__all__ = ...  # type: List[str]
+
+
+class weekday(object):
+    weekday = ... # type: int
+    n = ... # type: int
+    def __init__(self, weekday: int, n=...: Optional[int]) -> None: ...
```
Comment (matthiaskramm) [inline third_party/2.7/dateutil/relativedelta.pyi]: I think you meant `n: Optional[int] = ...`.
(Type annotations and default arguments are order-sensitive.)

Comment (rowillia) [inline third_party/2.7/dateutil/relativedelta.pyi]: @matthiaskramm Yup!  Thanks, fixed that up.

File: third_party/3/dateutil/relativedelta.pyi [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    def normalized(self) -> 'relativedelta': ...
+
+    def __add__(
+        self,
+        other: Union['relativedelta', timedelta, date, datetime]) -> 'relativedelta': ...
+
+    def __radd__(
+        self,
+        other: Any) -> 'relativedelta': ...
```
Comment (rowillia) [inline third_party/3/dateutil/relativedelta.pyi]: @matthiaskramm - Do you know if there is a better type to specify for `other`?  When I made it match the type for `__add__` I got:

```
third_party/2.7/dateutil/relativedelta.pyi: note: In member "__rsub__" of class "relativedelta":
third_party/2.7/dateutil/relativedelta.pyi:61: error: Forward operator "__sub__" is not callable
```

Comment (matthiaskramm) [inline third_party/3/dateutil/relativedelta.pyi]: Not sure. That error message looks odd.

@JukkaL: What does mypy mean, when it says that a function in a `pyi` is "not callable"?

Comment (gvanrossum) [inline third_party/3/dateutil/relativedelta.pyi]: Jukka is on vacation. It's not immediately obvious what this is, though there are a few known issues related to binary operators and their reverses. Can you boil it down to a stand-alone repro and file in the mypy tracker?

Comment (rowillia) [inline third_party/3/dateutil/relativedelta.pyi]: @gvanrossum Will do.  It's not 100% needed to merge this but a nice to have 😄

**Pull request merged**